flink-runtime/src/main/java/org/apache/flink/runtime/filecache/FileCache.java

```diff
@@ -99,7 +99,8 @@ public FileCache(String[] tempDirectories) throws IOException {
 		this.shutdownHook = createShutdownHook(this, LOG);
 
 		this.entries = new HashMap<JobID, Map<String, Tuple4<Integer, File, Path, Future<Path>>>>();
-		this.executorService = Executors.newScheduledThreadPool(10, ExecutorThreadFactory.INSTANCE);
+		this.executorService = Executors.newScheduledThreadPool(10, 
```

Comment:
This PR just did not want to change anything else than what its goal was.
The `10` is pretty magic, though, agreed. Something relative to the number of cores seems to make more sense, intuitively.